### PR TITLE
Add update strategy config option

### PR DESCRIPTION
**Pull request by ryanmcnamara**
values are "deferToPR" and "always"

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/bulldozer/30)

### CODE REVIEW

File: github/github.go [hunk 1]
```diff
@@ -56,6 +56,18 @@ var (
 	AcceptedPermLevels = []string{"write", "admin"}
 )
 
+type UpdateStrategy string
+
+const (
+	// UpdateStrategyDeferToPR the default value for UpdateStrategy
+	UpdateStrategyDeferToPR UpdateStrategy = "deferToPR"
```
Comment (ajlake) [inline github/github.go]: Could we do`UpdateStrategyLabel` and `"label"` for the names instead of `deferToPR`?

File: github/github.go [hunk 2]
```diff
@@ -119,6 +132,24 @@ func (client *Client) ConfigFile(repo *github.Repository, ref string) (*Bulldoze
...
+	validStrategy := func() bool {
+		for _, valid := range allowedUpdateStrategies {
+			if bulldozerFile.UpdateStrategy == valid {
+				return true
+			}
+		}
+		return false
+	}()
+	if !validStrategy {
+		return nil, errors.Errorf("Invalid update strategy %#v", bulldozerFile.UpdateStrategy)
```
Comment (ajlake) [inline github/github.go]: include the list of valid strategies in the error message? `allowedUpdateStrategies`

File: server/endpoints/hook.go [hunk 3]
```diff
@@ -85,16 +85,26 @@ func Hook(db *gorm.DB, secret string) echo.HandlerFunc {
...
 				}
-				if !updateLabel {
-					logger.WithFields(logrus.Fields{
-						"repo": repo.GetFullName(),
-						"pr":   pr.GetNumber(),
-					}).Info("Pull request does not have update me label")
-					continue
+
+				// Validate that we should update
+				switch repoConfig.UpdateStrategy {
```
Comment (ajlake) [inline server/endpoints/hook.go]: could we roll this check into a method?
```
for _, pr := range updateTargets {
  ...
  if ! shouldUpdatePR(...) {
    continue;
  }
  ...
  if pr.Head.Repo.GetFork() {
     ...
}
```

**Comment by ryanmcnamara** [comment]
fixes https://github.com/palantir/bulldozer/issues/21

File: server/endpoints/hook.go [hunk 4]
```diff
@@ -232,3 +226,18 @@ func Hook(db *sqlx.DB, secret string) echo.HandlerFunc {
 		return c.String(http.StatusOK, msg)
 	}
 }
+
+func shouldUpdatePR(logger *logrus.Entry, ghClient *gh.Client, repo *github.Repository, pr *github.PullRequest) (bool, error) {
+	updateLabel, err := ghClient.HasLabels(pr, config.UpdateMe)
+	if err != nil {
+		return false, errors.Wrapf(err, "cannot check if %s-%s has update label", repo.GetFullName(), pr.GetNumber())
```
Comment (ungureanuvladvictor) [inline server/endpoints/hook.go]: Change the format specifier to `%s-%d` `pr.GetNumber` returns an int https://github.com/google/go-github/blob/master/github/github-accessors.go#L5967.

**Pull request merged**